syntax = "proto3";
import "google/protobuf/wrappers.proto";

enum ActionResultStatus {
  SUCCESS = 0;
  ERROR = 1;
  BAD_ARGS = 2;
  CHECKSUM_SUM = 3;
  DOWNLOAD_ERROR = 4;
  FILE_FORMAT_ERROR = 5;
  WRITE_ERROR = 6;
  BAD_REQUEST_ERROR = 7;
}
message ActionResult {
  ActionResultStatus status = 1;
  oneof data {
    LatencyActionResult latency_action_result = 2;
  }
}

message LatencyActionResult {
    repeated LatencyProtobuf latency = 1;
}

message LatencyProtobuf {
    google.protobuf.StringValue prefix = 1;
    google.protobuf.StringValue timestamp = 2;
    google.protobuf.StringValue result = 3;
    google.protobuf.StringValue provider = 4;
    google.protobuf.StringValue region = 5;
    google.protobuf.StringValue dataCenter = 6;
    google.protobuf.StringValue target = 7;
    google.protobuf.StringValue targetIP = 8;
    google.protobuf.Int32Value avgRTT = 9;
    google.protobuf.Int32Value minRTT = 10;
    google.protobuf.Int32Value maxRTT = 11;
    google.protobuf.Int32Value stdRTT = 12;
    google.protobuf.Int32Value noSuccess = 13;
    google.protobuf.Int32Value noFailures = 14;
    google.protobuf.StringValue protocol = 15;
    google.protobuf.StringValue origin = 16;
    repeated google.protobuf.Int32Value rtts = 17;
    google.protobuf.StringValue errorCode = 18;
    google.protobuf.StringValue errorString = 19;
}
